packages/menu/src/test/mwc-menu.test.ts

```diff
@@ -0,0 +1,308 @@
+    const surface =
+        element.shadowRoot!.querySelector<MenuSurface>('.mdc-menu')!;
+    assert.equal(element.x, null);
+    assert.equal(element.y, null);
+    assert.equal(surface.x, null);
+    assert.equal(surface.y, null);
+    element.x = 111;
+    element.y = 101;
+    await element.updateComplete;
+    assert.equal(element.x, 111);
+    assert.equal(element.y, 101);
+    assert.equal(surface.x, 111);
+    assert.equal(surface.y, 101);
+  });
+
+  test('`absolute` is passed to surface', async () => {
+    const surface =
+        element.shadowRoot!.querySelector<MenuSurface>('.mdc-menu')!;
+    assert.equal(element.absolute, false);
+    assert.equal(surface.absolute, false);
+    element.absolute = true;
+    await element.updateComplete;
+    assert.equal(element.absolute, true);
+    assert.equal(surface.absolute, true);
+  });
+
+  test('`fixed` is passed to surface', async () => {
+    const surface =
+        element.shadowRoot!.querySelector<MenuSurface>('.mdc-menu')!;
+    assert.equal(element.fixed, false);
+    assert.equal(surface.fixed, false);
+    element.fixed = true;
+    await element.updateComplete;
+    assert.equal(element.fixed, true);
+    assert.equal(surface.fixed, true);
+  });
+
+  test('`fullwidth` is passed to surface', async () => {
+    const surface =
+        element.shadowRoot!.querySelector<MenuSurface>('.mdc-menu')!;
+    assert.equal(element.fullwidth, false);
+    assert.equal(surface.fullwidth, false);
+    element.fullwidth = true;
+    await element.updateComplete;
+    assert.equal(element.fullwidth, true);
+    assert.equal(surface.fullwidth, true);
+  });
+
+  test('correct roles are passed to inner list', async () => {
+    const list = element.shadowRoot!.querySelector<List>('.mdc-list')!;
+    assert.equal(element.innerRole, 'menu');
+    assert.equal(list.innerRole, 'menu');
+    assert.equal(list.itemRoles, 'menuitem');
+    element.innerRole = 'listbox';
+    await element.updateComplete;
+    assert.equal(list.innerRole, 'listbox');
+    assert.equal(list.itemRoles, 'option');
+  });
+
+  test('`multi` is set on inner list', async () => {
+    const list = element.shadowRoot!.querySelector<List>('.mdc-list')!;
+    assert.equal(element.multi, false);
+    assert.equal(list.multi, false);
+    element.multi = true;
+    await element.updateComplete;
+    assert.equal(element.multi, true);
+    assert.equal(list.multi, true);
+  });
+
+  test('`activatable` is set on inner list', async () => {
+    const list = element.shadowRoot!.querySelector<List>('.mdc-list')!;
+    assert.equal(element.activatable, false);
+    assert.equal(list.activatable, false);
+    element.activatable = true;
+    await element.updateComplete;
+    assert.equal(element.activatable, true);
+    assert.equal(list.activatable, true);
+  });
+
+  suite('selection', () => {
+    setup(async () => {
+      element.innerHTML = `
+        <mwc-list-item mwc-list-item>1</mwc-list-item>
```

Comment:
This seems weird. Readme shows using just a regular mwc-list-item. But the mdc sources specifically look for children with the _attribute_ mwc-list-item. So to make the test work I had to write it twice...

Reply:
This means that list item is not booting up. mwc-list-item-base applies this attribute on connectedCallback. You may have to wait for list item updatecomplete and menu updatecomplete. You may find some reference in the mwc-list tests

Reply:
using fixture in test helpers may help with this as well